### PR TITLE
Add android os

### PR DESCRIPTION
**Pull request by amfern**
Allthough android is linux the 'ip route' command doesn't return 'default' line.
Android has only one interface active at any time.

### CODE REVIEW

**Comment by amfern** [comment]
can someone give me a review?

File: ifaddrs.go [hunk 1]
```diff
@@ -1197,23 +1197,46 @@ func parseDefaultIfNameFromRoute(routeOut string) (string, error) {
...
+			parsedLine[3] == "dev" {
+			ifName := strings.TrimSpace(parsedLine[4])
+			return ifName, nil
+		}
+	}
+
+	return "", errors.New("No default interface found")
+}
+
+// parseDefaultIfNameFromIPCmd parses the default interface from ip(8) for
```
Comment (findkim) [inline ifaddrs.go]: can you update the function name with the suffix `Android`?

**Comment by amfern** [comment]
Thanks

**Comment by jefferai** [comment]
What's the reason to not support this? This is a library; Preeta's comment is about running an enterprise-grade scheduler on Nomad. I don't think they're equivalent.

**Comment by jefferai** [comment]
Reopening for further discussion.

File: ifaddrs.go (hunk 1 above)
Comment (ilyaiqoqo) [reply ifaddrs.go]: Thanks, Updated to `parseDefaultIfNameFromIPCmdAndroid`

**Comment by ilyaiqoqo** [comment]
Actually i agree, this is a library and not a nomad component, so it can have multiple applications.

**Comment by amfern** [comment]
Actually i agree, this is a library and not a nomad component, so it can have multiple applications.

File: ifaddrs.go (hunk 1 above)
Comment (amfern) [reply ifaddrs.go]: Thanks, Updated to parseDefaultIfNameFromIPCmdAndroid

**Comment by sean-** [comment]
Thank you for doing this @amfern, thank you for adding the bulk of this behind an android build tag .

**Comment by amfern** [comment]
Just checking in, what stops it from being merged, do you need something from me?

File: route_info_android.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package sockaddr
```
Comment (jefferai) [inline route_info_android.go]: I believe you need a build tag here.

Comment (amfern) [reply route_info_android.go]: I believe having `_android` suffix in the file name is sufficient enough, but added `// +build android` to be more explicit

Comment (jefferai) [reply route_info_android.go]: Ah you may be right, I don't deal with platform specific code much and forgot about that. But sure, being more explicit doesn't hurt!

Comment (jefferai) [reply route_info_android.go]: Actually, please remove it as there are already files in the package using the file name suffix pattern, so we might as well follow package convention.

Comment (jefferai) [reply route_info_android.go]: I removed it.

**Pull request merged**